src/NServiceBus.Core/EnumerableEx.cs

```diff
@@ -3,14 +3,13 @@ namespace NServiceBus
     using System;
     using System.Collections;
     using System.Collections.Generic;
-    using System.Linq;
     using Janitor;
 
     static class EnumerableEx
     {
         public static IEnumerable<T> Single<T>(T singleElement)
```

Comment:
> do we need this extension at all? why not just inline it?

I agree, not sure why we need an extension for this.

Looking at the rest of that class, I don't really care for any of the extension points there. If I'm looking at it correctly, using `EnsureNonEmpty` means that we'll always be allocating for enumeration, even when a collection has a non-allocating enumerator (for example, `List<t>`)

I would think changing anything about `EnsureNonEmpty` is out of scope for this PR though.